patches/tModLoader/Terraria/ModLoader/Config/ConfigManager.cs

```diff
@@ -356,6 +478,206 @@ public static bool EnumerableEquals(IEnumerable a, IEnumerable b)
 		}
 		return !hasNextA && !hasNextB;
 	}
+
+	internal static string FormatTextAttribute(string key, string localization, object[] args)
+	{
+		// TODO: support FindKeyInScope-type key shortening? 
+		if (args == null)
+			return localization;
+		for (int i = 0; i < args.Length; i++) {
+			if (args[i] is string s && s.StartsWith("$"))
+				args[i] = Language.GetTextValue(s.Substring(1));
+		}
+		return Language.GetText(key).Format(args);
+	}
+
+	// Used to determine which key to register, based only on field/property, not class, not necessarily which key to use in UI.
+	internal static string GetLabelKey(PropertyFieldWrapper memberInfo, ModConfig config, bool fallbackToClass, bool throwErrors)
+	{
+		var label = (LabelKeyAttribute)Attribute.GetCustomAttribute(memberInfo.MemberInfo, typeof(LabelKeyAttribute));
+		if (label != null) {
+			if (label.malformed && throwErrors)
+				throw new ValueNotTranslationKeyException($"{nameof(LabelKeyAttribute)} only accepts localization keys for the 'key' parameter.");
```

Comment:
An alternative to `malformed` has been provided in Discord, but on code-style, please refactor to reduce duplicate code as much as possible. It's hard to read when differences in behaviour are mixed in with common copy paste.

In this case 'get and validate' should be a method.

You can even simplify things like `(LabelKeyAttribute)Attribute.GetCustomAttribute(memberInfo.Type, typeof(LabelKeyAttribute))` if they start getting too long, like `GetAttr<LabelKeyAttribute>(memberInfo.Type)`

Ideally, a reader should be able to read your method and quickly understand exactly _what_ it does at a reasonable level of detail, thanks to good method names, without needing to understand _how_ it does it. If I want to know exactly _how_ you check attributes for malformed translations, I'll look into the `GetAndValidate` method.